### DISCUSSION TITLE
How can I switch the Github account associated with Github Copilot in Visual Studio?

### DISCUSSION THREAD

**Discussion by purplesoup** [community/community#68770] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I want to change the Github account associated with Github Copilot in Visual Studio.

I tried uninstalling and reinstalling the extension but I never had any prompt or option to re-assign the account. 

How do I switch accounts?

**Comment by josefdc** [comment]
To switch accounts for GitHub Copilot in Visual Studio, you may need to:

1. Remove Copilot from the extension list.
2. Reinstall GitHub Copilot.
3. Authorize with the desired account upon reinstallation.

It's also advised to clear your browser cache history before re-authorizing

**Reply by purplesoup** [reply]
Thanks for the reply - I thought this was what I had already done - what are you suggesting that is different?

**Reply by jzelAdmin2006** [reply]
Hi @purplesoup

To me it sounds like your browser is probably still signed in to the old account and it automatically authorizes Copilot when it opens the Browser window. You can sign out of GitHub in your browser or clear the history in order to get the login prompt. Let me know if that doesn't work or you have additional questions.

Best regards
JZEL

**Reply by purplesoup** [reply]
Thanks for the suggestion jzelAdmin2006

I had another go - I signed out of Github in my browser and uninstalled Copilot, close and re-opened Visual Studio, installed it again - but I never see any interaction with the browser - this is the output from the Copilot output when I sign in.

GitHub Copilot 1.116.0.0 (v1.116.0.0@5a26aa711)
GitHub Copilot Agent v1.116.0
[DEBUG] [agent] [2023-10-03T16:05:11.424Z] Agent service starting
[DEBUG] [agent] [2023-10-03T16:05:11.475Z] Telemetry initialized
Auth Status: OK
Successfully authorized GitHub Copilot.

This all happened while the browser was signed out and I never saw any interaction with a browser. It just seems to remember from somewhere my original Git account I used when I first installed it.

**Reply by DavySprocket** [reply]
I also did the same thing with no prompt.
I poked around and found a line in a config file with the account name but cannot edit it as it says its a system file.
The file is C:\Users\{user}AppData\Roaming\Code\User\globalStorage\globalState.json
line is: "github-DavySprocket": "[{\"id\":\"github.copilot\",\"name\":\"GitHub Copilot\",\"allowed\":true}]",

The line remains even though a extension removal.

**To get it to work I did this:**
1) close VSCode.
2) I opened C:\Users\{user}AppData\Roaming\Code\User\globalStorage\globalState.json in notepad and found and replaced every occurrence of my user name with nothing. 
3) Also deleted the backup file for good measure, not sure if this is nessisary.
4) opened VScode and got a prompt to sign in, in the bottom right.

Hope this helps.

**Comment by valkuznetsova** [comment]
I have the same issue and reinstalling neither VS nor CoPilot extension didn't help me.
And I don't have Sign Out option in Copilot Suggestions menu, it has "Add Account" and when I click it it throws unauthorized error due to my subscription expired.

**Reply by valkuznetsova** [reply]
Ok I was able to solve it in Account Settings
![image](https://github.com/community/community/assets/127252327/55af1465-583c-458f-b324-3d2e08e06a8a)

**Reply by DavySprocket** [reply]
Nice but Account settings does not exist in VS code.

**Reply by valkuznetsova** [reply]
Please note that this one works for VS, not VS Code
In VS Code you can find it in the left bottom corner
![image](https://github.com/community/community/assets/127252327/d80e4192-d131-460a-aa26-24143e3b5653)

**Reply by azegar** [reply]
thanks, it helps me as well

**Reply by ppeiwang** [reply]
thank for this info, it works

**Reply by melibermudez** [reply]
thanks! works on left bottom corner

**Reply by AjayHebburRavindra** [reply]
Thanks! It worked like a charm.

**Reply by chibuzoralumba** [reply]
This worked for me as well seamlessly

**Reply by apriharis** [reply]
Thanks, its work

**Reply by sistemasjah-25** [reply]
> Ok I was able to solve it in Account Settings ![image](https://private-user-images.githubusercontent.com/127252327/278139956-55af1465-583c-458f-b324-3d2e08e06a8a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI2MDY3NjgsIm5iZiI6MTc2MjYwNjQ2OCwicGF0aCI6Ii8xMjcyNTIzMjcvMjc4MTM5OTU2LTU1YWYxNDY1LTU4M2MtNDU4Zi1iMzI0LTNkMmUwOGUwNmE4YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEwOFQxMjU0MjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yYjczM2YzMzgwM2U2OTNjZDkwOTQxMThhODE5MDk1YTAzZWFmZmFlOGFmZTBlY2FlNTkxMGUwMWNkNDE3OThlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.1wR9vM7IaLpMCJhsGxdlRaPlW-VvITnmh_qH6qJojy0)

Great! Thanks

**Comment by DavySprocket** [comment]
To get it to work I did this:

1) close VSCode.
2) I opened C:\Users{user}AppData\Roaming\Code\User\globalStorage\globalState.json in notepad and found and replaced every occurrence of my user name with nothing.
3) Also deleted the backup file for good measure, not sure if this is nessisary.
4) opened VScode and got a prompt to sign in, in the bottom right.

Hope this helps.

**Comment by kratzl** [comment]
https://github.com/orgs/community/discussions/6823

**Comment by ayim** [comment, marked as answer]
There's two scenarios:

1) Within Tools > Options > GitHub > Copilot > Sign in with Device Flow >> **If this is FALSE**
![image](https://github.com/community/community/assets/10951869/b406e18a-c1fa-4e23-aafd-c74d080db4cf)
![image](https://github.com/community/community/assets/10951869/dbaf2ce5-3191-470f-9612-7da19497176e)

2) Within Tools > Options > GitHub > Copilot > Sign in with Device Flow >> **If this is TRUE** 
We recommend switching this setting to False, but if you must keep it true, then there is a Sign Out button in the Copilot menu.
![image](https://github.com/community/community/assets/10951869/58d986f4-47f7-4141-a87a-b0c79cf1764a)

**Reply by Amrit-synx** [reply]
where to find the tools option?

**Reply by mrbeandev** [reply]
here is the right solution for vscode --> 

> Click on extension Github Copilot. Then in settings Account Preference ![Screenshot 2024-12-21 at 12 24 37 PM](https://private-user-images.githubusercontent.com/153798410/397910406-437fa47a-831e-46ed-b139-2ab09730f918.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY3ODI1NDYsIm5iZiI6MTc1Njc4MjI0NiwicGF0aCI6Ii8xNTM3OTg0MTAvMzk3OTEwNDA2LTQzN2ZhNDdhLTgzMWUtNDZlZC1iMTM5LTJhYjA5NzMwZjkxOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwMlQwMzA0MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NmY1ZTZkZTgyNjYxZTg4M2RjN2UyOTI2MGUzMjJhYjZlN2QwYTA4ZmNlOTIwMDRhOThhMTQ3Y2Y5OGE0YjBhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.lZU1eZ-OpnxVnNHnPTZvg-6006556YR5KMVWvXbwVIk) It will show associated ones and/or option to add a new account. ![Screenshot 2024-12-21 at 12 25 35 PM](https://private-user-images.githubusercontent.com/153798410/397912387-ca54d835-735a-4741-a61b-345221f5fc2f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY3ODI1NDYsIm5iZiI6MTc1Njc4MjI0NiwicGF0aCI6Ii8xNTM3OTg0MTAvMzk3OTEyMzg3LWNhNTRkODM1LTczNWEtNDc0MS1hNjFiLTM0NTIyMWY1ZmMyZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwMlQwMzA0MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNDA3NDBkYzAxN2RmNDNiNmEyM2Q0MDFhYTQyYTA0MmNiOGNmZTYzM2VmZmQ4Yjk2M2NmNjQ0MTEzYWJlZTk5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.lvdKhAjXuigGzH1TBrHne3k0CXX-H6bq0V_4UmYMYH8)

**Reply by localhostd3veloper** [reply]
@mrbeandev  cant seem to see the images :)

**Comment by ashishMsci** [comment]
Nothing worked for me in vscode. Then I uninstalled the vscode and installed it in different location. That worked!

**Reply by HarisDev** [reply]
I had two GitHub accounts, but only one of them had Copilot associated. I resolved it by logging out of the unused one (located in the bottom left corner with the user icon).

You can also disable Github Copilot extension for specific Github account by clicking "Manage Trusted Extensions" and it should work.


![image](https://github.com/community/community/assets/12673108/e13eff35-01de-4a78-8bab-bbb52b694f1e)

**Reply by srea-credera** [reply]
That worked for me!

**Reply by de-buddy** [reply]
> Nothing worked for me in vscode. Then I uninstalled the vscode and installed it in different location. That worked!

uninstall extensions, sign out, reinstall extensions: That worked!

**Reply by auroch-forge** [reply]
<img width="749" height="265" alt="image" src="https://github.com/user-attachments/assets/08e44065-16ac-4b53-b30c-95a28da016a4" />

**Reply by Mr-Ketlhaetse** [reply]
@HarisDev Your solution worked and it didn't even take me a minute to execute. Thank you.

**Reply by IamLyvsain** [reply]
This should be the recommended solutions :)

**Comment by bismaya-pattnaik** [comment]
Click on extension Github Copilot. Then in settings Account Preference
![Screenshot 2024-12-21 at 12 24 37 PM](https://github.com/user-attachments/assets/437fa47a-831e-46ed-b139-2ab09730f918)
It will show associated ones and/or option to add a new account.
![Screenshot 2024-12-21 at 12 25 35 PM](https://github.com/user-attachments/assets/ca54d835-735a-4741-a61b-345221f5fc2f)

**Reply by ZhuangyingXu** [reply]
This works and seems to be the easiest solution!

**Reply by brandonkam12** [reply]
this solution worked for me, thanks for the help! :)

**Reply by nikhilsaini007** [reply]
this works well :)

**Reply by Timilehin2509** [reply]
I found this out by accident just a few minutes ago lol

**Reply by ManchemVishnuSrikar** [reply]
easiest way!! thank you!!

**Comment by Sebast38** [comment]
Hello

Solution that worked for me : 

Go to your Windows Credentials Identity interface
Then delete all github ones
Then re-open Visual Studio

Hope it helps :)

Sebastien.

**Comment by danielpuertaortiz** [comment]
Just in the case someone else is having troubles in 2026. I had to open the Command Palette (CMD + shift + p on Mac, Ctrl + shift + p on windows)  -> Look for Accounts: Manage Extension Preferences -> Select Github Copilot Account -> Add another account

**Reply by escamlo101-arch** [reply]
thanks

**Reply by stillhart** [reply]
works for me. The account ordering suddenly randomly changed downgrading me to free copilot. thanks.
